src/tests/system/tests/test_passkey.py

```diff
@@ -0,0 +1,480 @@
+    client.sssd.start()
+
+    output2 = client.auth.passkey.su(tc, username, 123456)
+    assert f'su: user {username} does not exist' in output2.stdout_lines[-1] and output1.stdout_lines[-1] == 'user1'
+
+
+@pytest.mark.tier(1)
+@pytest.mark.topology(KnownTopology.IPA)
+@pytest.mark.topology(KnownTopology.LDAP)
+def test_passkey__su_fail_when_srv_offline_firewalld(client: Client, provider: GenericProvider):
+    """
+    :title: Check authentication deny of the user with LDAP and IPA from cache when server is unreachable
+    :setup:
+        1. Add LDAP and IPA user with passkey_mapping.
+        2. Setup IDM client with FIDO and umockdev setup
+        3. Create the recording files when authentications is successful
+        4. Add 'pam_passkey_auth = True' under pam section of sssd.conf file.
+    :steps:
+        1. Check the umockdev-run command for authentication of the user.
+        2. Check umockdev-run that the output contains the username.
+        3. Remove the cache and make server unreachable by adding firewalld rule to remove ldap and ldaps ports.
+        4. Check the authentication of user
+    :expectedresults:
+        1. Command returns 0.
+        2. Output contains username of the user.
+        3. Successfully remove the cache and made server offline.
+        4. Authentication deny getting error message.
+    :customerscenario: False
+    """
+    tc = passkey_str = username = ""
+    if isinstance(provider, IPA):
+        tc = "su_ipa_test3"
+        username = 'user1@ipa.test'
+        passkey_str = "passkey:NUZMRUXIb/W8Ij1GqwCDHSCWxt/SxWxckwtQjLYi/X6Y1qZFB+HI8WO6khzAjzsz248" \
+                      "kHbaeAf9qfmqfCky1Jg==,MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEIasAa8ogjPCKXeA4K" \
+                      "Y3t0W3xBRmG+E4D+MNoRIAJrYuNLSYtAcOL7DCbIfgc+7c5Y4Mh/FzoEyeumKGYMoyTfg=="
+
+    elif isinstance(provider, LDAP):
+        tc = "su_ldap_test6"
+        username = "user1@test"
+        passkey_str = "passkey:mQEUTWdtDJPELQNTDdxXNHlfIO1qXFf0LVZjWEfyDALFzvLZ4e4XD5bemqq+o3ThrzT6k1I1n3Z2N00G" \
+                      "vLSmjQ==,MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEqk7K5VAI7Evr4ar8X82L/sxm/Bnm5Ti31xnLfGO0BipwH" \
+                      "ucw8+/wT4+6T9j5gdMwZKUcXR4BILpmULEyrcZUfw=="
+
+    user = provider.user('user1').add()
+    user.passkey_add(passkey_str)
+
+    client.sssd.pam['pam_passkey_auth'] = 'true'
+    client.sssd.start()
+
+    for record_file in [f'{tc}.ioctl', f'{tc}.script', f'{tc}.device']:
+        src_file = f"{os.getcwd()}/data/passkey/{tc}/{record_file}"
+        client.fs.upload(src_file, f"/tmp/{record_file}")
+
+    output1 = client.auth.passkey.su(tc, 'user1@test', 123456)
+
+    for cmd in ['dnf install firewalld -y', 'systemctl start firewalld',
+                'firewall-cmd --permanent --remove-port=636/tcp',
+                'firewall-cmd --permanent --remove-port=389/tcp', 'firewall-cmd --reload']:
+        provider.host.ssh.run(cmd, raise_on_error=False)
```

Comment:
Please, do not install packages in tests. This needs to be added to sssd-ci-containers and firewall-cmd should be turned to utility class. I will work on that.

Reply:
I remove the above test cases as of from PR.